lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -312,28 +311,18 @@ _estimateMaxNumberOfInputs
     -- ^ Number of outputs in transaction
     -> Word8
 _estimateMaxNumberOfInputs networkId (Quantity maxSize) md nOuts =
-      fromIntegral $ bisect (lowerBound, upperBound)
+    findMax minBound
   where
-    bisect (!inf, !sup)
-        | middle == inf && isTooBig sup = inf
-        | middle == inf                 = sup
-        | isTooBig middle               = bisect (inf, middle)
-        | otherwise                     = bisect (middle, sup)
-      where
-        middle = inf + ((sup - inf) `div` 2)
-
-    growingFactor = 2
-
-    lowerBound = upperBound `div` growingFactor
-    upperBound = upperBound_ 1
-      where
-        upperBound_ !n | isTooBig n = n
-                       | otherwise  = upperBound_ (n*growingFactor)
+    findMax :: Word8 -> Word8
+    findMax inf
+        | inf == maxBound    = 0
```

Comment:
What does `inf == maxBound` mean? If `1` input would be too much, `findMax 0` should return 0. But it _also_ returns 0 if all possibilities pass? That seems like an error. (If the tx size is really big...)

Reply:
Actually you're right, this should just return `maxBound` :+1: